src/codegate/pipeline/base.py

```diff
@@ -275,6 +275,13 @@ def __init__(
         self.secret_manager = secret_manager
         self.is_fim = is_fim
         self.context = PipelineContext()
+
+        # we create the sesitive context here so that it is not shared between individual requests
+        # TODO: could we get away with just generating the session ID for an instance?
```

Comment:
Assuming each request has exactly one instance of the pipeline and each pipeline has one instance of PipelineSensitiveData, the session ID can be generated for the pipeline instance and used in PipelineSensitiveData.